### DISCUSSION TITLE
Error on page ?

### DISCUSSION THREAD

**Discussion by StanMOI** [sanni/cartreader#993] [General Chat]
Hi Sanni,

Must be some error on this page. It should be "4MB" , isn't it? :

[https://github.com/sanni/cartreader/wiki/Reflashing-SNES-SFC-Repros](https://github.com/sanni/cartreader/wiki/Reflashing-SNES-SFC-Repros)
<img width="937" alt="Capture d’écran 2024-07-19 à 23 48 07" src="https://github.com/user-attachments/assets/1af647d2-49e8-485f-9825-fadff69f1b46">

Can I suggest you if I can add text to this Manual for how to use this option for flashing EX-HiROM or EX-LoROM ?

Use two Options **P0** and **P1** for Writing flashrom with any file more then 4MB.
Here is Exemple for 6MB :

1. Split file into two files. 
You can use [Advanced SNES ROM Utility](https://www.romhacking.net/utilities/1638/)

- Via **Advanced SNES ROM Utility** Load original file (in this exemple 6MB). 
- Chose "Split ROM" and select "32Mbit (4MByte) 27C322"
- Click to "Split ROM" button.
- You will get two splitted files : one file with 4MB size, second file with 2MB.

2. Cart Reader: Select P0, erase the chip and flash the 2MB file.
3. Switch Cart Reader off/on, select P1, do not erase again, and flash the 4MB file.

**Comment by sanni** [comment]
LoRom (P0) can flash 8MB if its a 29LV640 repro. It's the same function as LoRom (P1) just that (P1) flips A22. 
So with P0 the flashrom gets written 12345678.
And with P1 it gets written 56781234.
Where each number represents a 1MB part of the 8MB file that gets written.

**Reply by StanMOI** [reply]
Why we can not flash 8MB with HiRom (P0)?

**Reply by sanni** [reply]
I don't have a hirom repro with 29LV640, so I can't test if 8MB flashing works. If someone tests it we can update the Wiki.

**Reply by StanMOI** [reply]
I’ll do it later.
I got my own PCB so I can do one special for this test.

**Reply by StanMOI** [reply]
> I don't have a hirom repro with 29LV640, so I can't test if 8MB flashing works. If someone tests it we can update the Wiki.

Cannot test 8MB HiROM with P0.
It's says "File too big"

**Comment by StanMOI** [comment]
Writing test for **P0** and **CFI** options:
**Flash S29GL064**

LoROM (P0)
ROM File size: 4MB 
Write / Verif / SNES test= OK

——
CFI LoROM 
1x 4MB
ROM File size: 4MB 
Write / Verif / SNES test= OK

——
LoROM (P0)
Game name "Fire Emblem" (8MB exLoROM-SRAM 256)
ROM File size: 8MB 
Write / Verif = OK
SNES test= KO

——
CFI LoROM 
1x 4MB
ROM File size: 8MB 
Write = File too big

+++++++++++
Writing test for P0 and CFI options:
**Flash S29GL064**

HiROM (P0)
ROM File size: 4MB 
Write / Verif / SNES test= OK

——
CFI HiROM 
1x 4MB
ROM File size: 4MB 
Write / Verif / SNES test= OK

——
HiROM (P0)
ROM File size: 8MB 
Write = File too big

——
CFI HiROM 
1x 4MB
ROM File size: 8MB 
Write = File too big

**Reply by sanni** [reply]
This version has a 1x8MB option for CFI under SNES flash repro. Untested.
https://github.com/sanni/cartreader/discussions/963#discussioncomment-10241449

**Reply by StanMOI** [reply]
> This version has a 1x8MB option for CFI under SNES flash repro. Untested. [#963 (reply in thread)](https://github.com/sanni/cartreader/discussions/963#discussioncomment-10241449)

Tests CFI with 1x 8MB done.
Not working for me.
Erasing, Writing, Verifying done successfully but on SNES I get black screen for both functions (standard and reversed).

![IMG_7826](https://github.com/user-attachments/assets/0067848b-8725-435d-aa97-c4cfdfc56cf6)
![IMG_7823](https://github.com/user-attachments/assets/e40c8cc3-d661-49f7-a988-66207693d331)

![IMG_7825](https://github.com/user-attachments/assets/c328be09-4b63-48a1-8bee-7da49d12e6f3)
![IMG_7827](https://github.com/user-attachments/assets/9ebb3c0e-d13a-4a08-824f-0a33b8b7d1e8)

**Comment by StanMOI** [comment]
Sanni, 
I done tests with the same PCB and Flash S29GL064.
As you see, 
when I try to write 8MB rom with function "LoROM (P0)", all OK (Write / Verif ) except SNES test (I get black screen).
When I do test with the same rom but splitted 8MB file into 2X4MB and using functions P0 and P1 to write, then SNES test works. 
So I tried to write 8MB directly with function "ExLoRom (P1)" , result = ALL OK (Write / Verif / SNES test).
Do i do something wrong ?

[- LoRom (P0) will write up to 8MB starting from file 0x0 to flashrom 0x0.](https://github.com/sanni/cartreader/wiki/Reflashing-SNES-SFC-Repros)

**Reply by sanni** [reply]
LoROM (P0) writes 12345678 and ExLoRom (P1) writes 56781234. So your game probably needs 56781234 order.

**Reply by StanMOI** [reply]
> LoROM (P0) writes 12345678 and ExLoRom (P1) writes 56781234. So your game probably needs 56781234 order.

What makes me confuse is that says for LoROM (P0) "will write up to 8MB".
I done tests with LoROM (P0) for 4MB, 6MB and 8MB  without success, accept 4MB.
But for ExLoRom (P1) writing 8MB works.

Has anyone done successfull tests for LoROM (P0) and 8MB file?
